src/hook.c

```diff
@@ -757,3 +761,46 @@ MH_STATUS WINAPI MH_ApplyQueued(VOID)
         LeaveSpinLock();
     }
 }
+
+//-------------------------------------------------------------------------
+MH_STATUS WINAPI MH_CreateHookApi(LPCSTR pszTarget, LPVOID pDetour, LPVOID *ppOriginal)
+{
+    HMODULE hModule;
+    LPVOID  pTarget;
+    SIZE_T  copySize;
+    LPCSTR  psz = pszTarget;
+    LPCSTR  pszProcName = NULL;
+    CHAR    szModuleName[MAX_PATH];
+
+    if (pszTarget == NULL || pDetour == NULL)
+        return MH_ERROR_INVALID_ARGUMENTS;
+
+    // search the last '.' (replacement for strrchr())
+    while (*psz)
+        if (*psz++ == '.')
+            pszProcName = psz;
```

Comment:
I haven't tried the code, but it looks like it searches for the first '.' char. Therefore, an input like "user32.dll.MessageBoxW" won't work correctly, would it?
Also, I personally don't find this function very helpful for MinHook, sorry. In my opinion, it doesn't ease the hooking procedure too much. It's helpful mostly when an exported function doesn't exist in the headers, and even then, it merely saves two calls: `GetModuleHandle` and `GetProcAddress`. (and even then it's not complete as `GetModuleHandleA` doesn't accept Unicode file names)